llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -3177,10 +3178,10 @@ InstructionCost AArch64TTIImpl::getCFInstrCost(unsigned Opcode,
   return 0;
 }
 
-InstructionCost AArch64TTIImpl::getVectorInstrCostHelper(const Instruction *I,
-                                                         Type *Val,
-                                                         unsigned Index,
-                                                         bool HasRealUse) {
+InstructionCost AArch64TTIImpl::getVectorInstrCostHelper(
+    Type *Val, unsigned Index, bool HasRealUse, const Instruction *I,
+    std::optional<unsigned> Opcode, Value *Scalar,
```

Comment:
>The code to call it can drop the `if (I || Opcode)`

This check needs to be done because it segfaults when `Scalar=nullptr` while doing cast<ExtractelemementInst>. So, either the check remains or needs to be pushed inside. I have kept it outside.